lms/product/product.py

```diff
@@ -39,9 +42,12 @@ def __post_init__(self, product_settings):
 
 
 @dataclass
-class Product:
+class Product:  # pylint:disable=too-many-instance-attributes
     """The main product object which is passed around the app."""
 
+    request: IRequest
```

Comment:
Let's keep a reference of the request here.

This is accessed via `request.product`.

 It would be silly to have to do request.product.api_client(request) IMO.

Reply:
I assume there is never a scenario in which we instantiate a `Product` without having a request available?